typescript/src/timestamp.ts

```diff
@@ -43,7 +54,12 @@ export function googleProtobufTimestampFromProto3JSON(json: string) {
   const date = new Date(json);
   const millisecondsSinceEpoch = date.getTime();
   const seconds = Math.floor(millisecondsSinceEpoch / 1000);
-  const nanos = (millisecondsSinceEpoch % 1000) * 1000000;
+  // The fractional seconds in the JSON timestamps can go up to 9 digits (i.e. up to 1 nanosecond resolution).
+  // However, Javascript Date object represent any date and time to millisecond precision.
+  // To keep the precision, we extract the fractional seconds and append 0 until the length is equal to 9.
+  const nanos = parseInt(
+    (json.split('.')[1].slice(0, -1) + '000000000').slice(0, 9)
```

Comment:
That looks cool :)  Instead of `+ '000000000') and slicing, let's do `.padEnd(9, '0')`, then it will have exactly 9 characters.

Reply:
Great.